darts/models/forecasting/torch_forecasting_model.py

```diff
@@ -883,7 +904,7 @@ def _train(self,
             total_loss = 0
 
             for batch_idx, train_batch in enumerate(train_loader):
-                self.model.train()
+                self.__set_train_mode()
```

Comment:
Do we need `__set_train_mode()`? Since it just calls `model.train()` I believe we can just leave it as before?

Concerning `__set_eval_mode()` we can also customize this  with PyTorch Lightning's `on_predict_model_eval()`